tests/wallet/sync/test_wallet_sync.py

```diff
@@ -27,209 +27,207 @@ def wallet_height_at_least(wallet_node, h):
+            await time_out_assert(600, wallet_height_at_least, True, wallet_node, len(default_1000_blocks) - 1)
 
-        await disconnect_all_and_reconnect(wallet_server, full_node_server, self_hostname)
+            await disconnect_all_and_reconnect(wallet_server, full_node_server, self_hostname)
 
         # Tests a short reorg
         num_blocks = 30
         blocks_reorg = bt.get_consecutive_blocks(num_blocks, block_list_input=default_1000_blocks[:-5])
 
-        for i in range(1, len(blocks_reorg)):
+        for i in range(len(blocks_reorg) - num_blocks - 10, len(blocks_reorg)):
             await full_node_api.full_node.respond_block(full_node_protocol.RespondBlock(blocks_reorg[i]))
 
-        await time_out_assert(
-            600, wallet_height_at_least, True, wallet_node, len(default_1000_blocks) + num_blocks - 5 - 1
-        )
+        for wallet_node, wallet_server in wallets:
+            await time_out_assert(
+                600, wallet_height_at_least, True, wallet_node, len(default_1000_blocks) + num_blocks - 5 - 1
+            )
 
-    @pytest.mark.parametrize(
-        "trusted",
-        [True, False],
-    )
     @pytest.mark.asyncio
-    async def test_wallet_reorg_sync(self, bt, wallet_node_simulator, default_400_blocks, trusted, self_hostname):
+    async def test_wallet_reorg_sync(self, bt, two_wallet_nodes, default_400_blocks, self_hostname):
         num_blocks = 5
-        full_nodes, wallets = wallet_node_simulator
+        full_nodes, wallets = two_wallet_nodes
         full_node_api = full_nodes[0]
-        wallet_node, server_2 = wallets[0]
-        fn_server = full_node_api.full_node.server
-        wsm: WalletStateManager = wallet_node.wallet_state_manager
-        wallet = wsm.main_wallet
-        ph = await wallet.get_new_puzzlehash()
+        full_node_server = full_node_api.full_node.server
+
+        # Trusted node sync
+        wallets[0][0].config["trusted_peers"] = {full_node_server.node_id.hex(): full_node_server.node_id.hex()}
 
-        if trusted:
-            wallet_node.config["trusted_peers"] = {fn_server.node_id.hex(): fn_server.node_id.hex()}
-        else:
-            wallet_node.config["trusted_peers"] = {}
+        # Untrusted node sync
+        wallets[1][0].config["trusted_peers"] = {}
 
-        await server_2.start_client(PeerInfo(self_hostname, uint16(fn_server._port)), None)
+        phs = []
+        for wallet_pair in wallets:
+            wallet_node, wallet_server = wallet_pair
+            wallet = wallet_node.wallet_state_manager.main_wallet
+            phs.append(await wallet.get_new_puzzlehash())
+            await wallet_server.start_client(PeerInfo(self_hostname, uint16(full_node_server._port)), None)
 
         # Insert 400 blocks
         for block in default_400_blocks:
             await full_node_api.full_node.respond_block(full_node_protocol.RespondBlock(block))
 
         # Farm few more with reward
+        for i in range(0, num_blocks - 1):
```

Comment:
Why `- 1`?

Reply:
The test was testing only one wallet, but farming rewards take 1 block to get confirmed. So here the first wallet needs less rewards because all of them will get confirmed, in the second one, the last one will not get confirmed

Reply:
Thanks for the explanation.  I think I might generally prefer to make sure all the rewards get processed rather than finishing the test with unconfirmed transactions and off-by-one compensations.

But, you can go ahead and resolve this conversation once you've read my comment.  No action requested.